### ISSUE TITLE
[BUG] Custom field "Link to document" takes forever

### ISSUE THREAD

**Issue by hspe98** [paperless-ngx/paperless-ngx#5011]
### Description

If I want to link a document to another document using the new "custom fields" feature, the suggestions come relatively quickly in the drop-down menu, but when I select an item from the list, a spinning wheel appears on the right side of the box and the link only appears after waiting 1-2 minutes. When adding a second link for a document, it works directly.

### Steps to reproduce

1. New custom field "Dokumentenverknüpfung"
2. Assign this field to a document
3. search for a document in the box

### Webserver logs

```bash
webserver_1  | /usr/local/lib/python3.11/site-packages/rest_framework/pagination.py:200: UnorderedObjectListWarning: Pagination may yield inconsistent results with an unordered object_list: <class 'documents.models.Document'> QuerySet.
webserver_1  |   paginator = self.django_paginator_class(queryset, page_size)
webserver_1  | /usr/local/lib/python3.11/site-packages/django/http/response.py:517: Warning: StreamingHttpResponse must consume synchronous iterators in order to serve them asynchronously. Use an asynchronous iterator instead.
webserver_1  |   warnings.warn(
webserver_1  | /usr/local/lib/python3.11/site-packages/rest_framework/pagination.py:200: UnorderedObjectListWarning: Pagination may yield inconsistent results with an unordered object_list: <class 'documents.models.Document'> QuerySet.
webserver_1  |   paginator = self.django_paginator_class(queryset, page_size)
webserver_1  | [2023-12-17 11:10:28,539] [INFO] [celery.app.trace] Task paperless_mail.tasks.process_mail_accounts[c48e4dc9-f8d5-45ba-9004-5b0152778aa4] succeeded in 28.515634324008715s: 'No new documents were added.'
webserver_1  | /usr/local/lib/python3.11/site-packages/rest_framework/pagination.py:200: UnorderedObjectListWarning: Pagination may yield inconsistent results with an unordered object_list: <class 'documents.models.Document'> QuerySet.
webserver_1  |   paginator = self.django_paginator_class(queryset, page_size)
webserver_1  | /usr/local/lib/python3.11/site-packages/django/http/response.py:517: Warning: StreamingHttpResponse must consume synchronous iterators in order to serve them asynchronously. Use an asynchronous iterator instead.
```

### Paperless-ngx version

2.1.2

### Host OS

Raspberry Pi4 aarch64

### Installation method

Docker - official image

### Browser

Chrome

### Configuration changes

-

**Comment by shamoon** [comment]
I can't reproduce this. Are there any errors in the browser? Can you see in the browser network tab if the calls to the API are taking a very long time? How many documents do you have?

**Comment by shamoon** [comment]
Even though I cant reproduce it I think some improvements would help, if you have a second to test out the docker image `feature-5011-get-few` let me know if that does it.

FYI, technically that image will be based of 2.1.3 (your post says your on 2.1.2), even though Im not aware of anything that should affect your install permanently (i.e. there are no db migrations between those two versions see https://github.com/paperless-ngx/paperless-ngx/compare/v2.1.2...v2.1.3 ), I will say its always safest to test with a fresh install or to at least have a backup in case

**Comment by hspe98** [comment]
Hello,

I have around 2.127 documents in paperless-ngx. I've recently upgrade from an 32-bit-installation (1.17) on a raspberry pi to a 64-bit installation (2.1.2). I've moved the folders with documents and exported the postgresql database to a dump, which I've imported to the database before I started the paperless-ngx docker-compose. I've have to tweak a few tables which seems to not have been correctly imported from the dump (like `auth_user`). But now the database is fine an paperless-ngx is running very fast compared to the previous installation.

I've tried to reproduce with a different document:

This request takes forever:
`https://paperless.XXX.yyy/api/documents/?page=1&page_size=100000
`

This is from the Timing tab in the Chrome developer tools (unfortunately in German, sorry):
![image](https://github.com/paperless-ngx/paperless-ngx/assets/7887268/19c90697-7629-4d89-8d92-34b753bba327)

The 39 seconds are roughly the time it takes to finally show the document link in the box. When I try to query the url from browser (`/api/documents/?format=json&page=1&page_size=XXX`) and increment the value of page_size (`XXX`), I get following timing for each value:

```
page_size=10         1,23s
=100                 2,79s
=500                10,25s
=1000               23,07s
```

When I use `&truncate_content=true`, the time for `page_size=1000` decrease to 17,44 seconds (-25%).
